x/group/server/invariants.go

```diff
@@ -169,3 +182,91 @@ func groupTotalWeightInvariant(ctx sdk.Context, groupTable orm.Table, groupMembe
 	}
 	return msg, broken, err
 }
+
+func proposalTallyInvariant(ctx sdk.Context, proposalTable orm.AutoUInt64Table, groupAccountTable orm.PrimaryKeyTable, groupMemberByGroupIndex orm.UInt64Index, voteTable orm.PrimaryKeyTable) (string, bool, error) {
+	var msg string
+	var broken bool
+
+	var proposal group.Proposal
+	var groupAcc group.GroupAccountInfo
+	var groupMem group.GroupMember
+	var vote group.Vote
+
+	proposalIt, err := proposalTable.PrefixScan(ctx, 1, math.MaxUint64)
+	if err != nil {
+		return msg, broken, err
+	}
+	defer proposalIt.Close()
+
+	membersWeight := apd.New(0, 0)
+
+	for {
+		_, err := proposalIt.LoadNext(&proposal)
+		if orm.ErrIteratorDone.Is(err) {
+			break
+		}
+		address, err := sdk.AccAddressFromBech32(proposal.Address)
+		if err != nil {
+			return msg, broken, err
+		}
+		err = groupAccountTable.GetOne(ctx, address.Bytes(), &groupAcc)
+		if err != nil {
+			return msg, broken, err
+		}
+		groupMemIt, err := groupMemberByGroupIndex.Get(ctx, groupAcc.GroupId)
+		if err != nil {
+			return msg, broken, err
+		}
+		defer groupMemIt.Close()
+
+		for {
+			_, err := groupMemIt.LoadNext(&groupMem)
+			if orm.ErrIteratorDone.Is(err) {
+				break
+			}
+
+			voterAddress, err := sdk.AccAddressFromBech32(groupMem.Member.GetAddress())
+			if err != nil {
+				return msg, broken, err
+			}
+
+			err = voteTable.GetOne(ctx, group.Vote{ProposalId: proposal.ProposalId, Voter: voterAddress.String()}.PrimaryKey(), &vote)
+			if err != nil {
+				return msg, broken, err
+			}
+
+			curMemWeight, err := regenMath.ParseNonNegativeDecimal(groupMem.GetMember().GetWeight())
+			if err != nil {
+				return msg, broken, err
+			}
+
+			voteChoice := strconv.FormatInt(int64(group.Choice_value[vote.GetChoice().String()]), 10)
+			curVoteWeight, err := regenMath.ParseNonNegativeDecimal(voteChoice)
+			if err != nil {
+				return msg, broken, err
+			}
+
+			if curMemWeight.Cmp(curVoteWeight) != 0 {
```

Comment:
this is wrong, the member's weight is not supposed to be equal to the vote choice, that would mean that a member with weight 2 would always have to vote yes (because `Choice_CHOICE_YES Choice = 2`).

Reply:
okay, I'll make necessary changes